posthog/settings/data_stores.py

```diff
@@ -101,9 +101,16 @@
 CLICKHOUSE_HTTP_URL = f"{_clickhouse_http_protocol}{CLICKHOUSE_HOST}:{_clickhouse_http_port}/"
 
 # Kafka configs
+
+_parse_kafka_hosts = lambda kafka_url: ",".join(urlparse(host).netloc for host in kafka_url.split(","))
+
+# URL Used by kafka clients/producers
 KAFKA_URL = os.getenv("KAFKA_URL", "kafka://kafka")
-KAFKA_HOSTS_LIST = [urlparse(host).netloc for host in KAFKA_URL.split(",")]
-KAFKA_HOSTS = ",".join(KAFKA_HOSTS_LIST)
+KAFKA_HOSTS = _parse_kafka_hosts(KAFKA_URL)
+
+# Kafka broker host(s) that is used by clickhouse for ingesting messages. Useful if clickhouse is hosted outside the cluster.
+KAFKA_HOSTS_CLICKHOUSE = _parse_kafka_hosts(os.getenv("KAFKA_URL_CLICKHOUSE", KAFKA_URL))
```

Comment:
What about `KAFKA_URL_EXTERNAL`?

Reply:
Not used anywhere hence thought best not to expose it.

Reply:
From discussion: `KAFKA_HOSTS_FOR_CLICKHOUSE` and similar name for the env var :)